### PR TITLE
Preferences Lv2: fix parameter list

### PR DESCRIPTION
**Pull request by ronso0**
https://bugs.launchpad.net/mixxx/+bug/1786361

image

Also the effect names are now aligned left so that the list is easier to browse and the labels are not obscured that easy when the horizontal scrollbars are shown.

### CODE REVIEW

**Comment by ronso0** [comment]
**ToDo**: fix Tab navigation.
Currently both Tab/Shift+Tab and up/down keys can navigate through both the button column and the parameter list, so in order to get back to the sidebar one has to go through the entire list.

Solution: KeyEvent filter so that
Tab jumps between columns
Up/Down moves focus in the lists

**Comment by ronso0** [comment]
> **ToDo**: fix Tab navigation.
> Currently both Tab/Shift+Tab and up/down keys can navigate through both the button column and the parameter list, so in order to get back to the sidebar one has to go through the entire list.
> 
> Solution: KeyEvent filter so that
> Tab jumps between columns
> Up/Down moves focus in the lists

I tried that but it requires some ugly hacks.
Alternative would be to build the effect list and the parameter list like in #2618 but that's too much for this little PR trying to improve regular navigation.

**Comment by Holzhaus** [comment]
Any free LV2 effects you can recommend? I don't have any installed atm, but I guess I need some for testing.

**Comment by Holzhaus** [comment]
Also, can you add a changelog entry? Or do you think this doesn't need one?

**Comment by ronso0** [comment]
I have a bunch installed but never use them actually...
try `eq10q` or `guitarix-lv2`.

Don't think this is worth a changelog entry.

File: src/preferences/dialog/dlgpreflv2.cpp [hunk 1]
```diff
@@ -108,7 +113,7 @@ void DlgPrefLV2::slotDisplayParameters() {
     } else {
         m_iCheckedParameters = 0;
     }
-    lv2_vertical_layout_params->addStretch();
+    lv2_VLayout_parameters->addStretch();
```
Comment (Holzhaus) [inline src/preferences/dialog/dlgpreflv2.cpp:116]: Let's either just camel case or underscores, not both.

File: src/preferences/dialog/dlgpreflv2.cpp [hunk 2]
```diff
@@ -87,18 +88,22 @@ void DlgPrefLV2::slotDisplayParameters() {
 
     EffectManifestPointer pCurrentEffectManifest = m_pLV2Backend->getManifest(pluginId);
     if (pCurrentEffectManifest) {
+        // Show the effect name above the parameter list
+        effect_name_label->setText(QObject::tr("Parameters for %1")
```
Comment (Holzhaus) [inline src/preferences/dialog/dlgpreflv2.cpp:92]: camel case?

**Review by Holzhaus — CHANGES_REQUESTED** [review]

File: src/preferences/dialog/dlgpreflv2.cpp (hunk 2 above)
Comment (ronso0) [reply src/preferences/dialog/dlgpreflv2.cpp:92]: in the preferences ui files I couldn't figure a rule.
I'll go with _ for better readability.

**Comment by ronso0** [comment]
Done.

File: src/preferences/dialog/dlgpreflv2.cpp [hunk 3]
```diff
@@ -76,9 +78,8 @@ void DlgPrefLV2::slotDisplayParameters() {
     m_pluginParameters.clear();
 
     QLayoutItem* item;
-    while ((item = lv2_vertical_layout_params->takeAt(1)) != 0) {
-        lv2_vertical_layout_params->removeWidget(item->widget());
-        delete item->widget();
+    while ((item = lv2EffectParametersList->takeAt(0)) != 0) {
+        delete item;
```
Comment (Holzhaus) [inline src/preferences/dialog/dlgpreflv2.cpp:82]: I'm not a fan of assignments in condition expressions. 

```suggestion
    while (! lv2EffectParametersList->isEmpty()) {
        QLayoutItem* item = lv2EffectParametersList->takeAt(0);
        VERIFY_OR_DEBUG_ASSERT(item) {
            continue;
        }
        delete item;
} 
```

**Review by Holzhaus — CHANGES_REQUESTED** [review]

File: src/preferences/dialog/dlgpreflv2.cpp (hunk 3 above)
Comment (ronso0) [reply src/preferences/dialog/dlgpreflv2.cpp:82]: here `isEmpty()` doesn't count spacers which would remain at index 0 after having deleted all _real_ items.
`count()` does the trick :+1:

**Comment by ronso0** [comment]
Ready.

**Review by Holzhaus — APPROVED** [review]
LGTM, thank you.

**Comment by Holzhaus** [comment]
CI failure is unrelated.

**Pull request merged**